controllers/feeds/reactionList.js

```diff
@@ -14,8 +15,21 @@ module.exports = async (req, res) => {
     const {params, query} = req;
     const post = await Getstream.feed.getPlainFeedById(params.id);
     const myAnonymousId = await getAnonymUser(req.userId);
+    const blockedUserList = await getListBlockUser(req.userId);
+    // Convert string to array if necessary
+    let blockedUserArray;
+    if (typeof blockedUserList === 'string') {
+      try {
+        blockedUserArray = JSON.parse(blockedUserList);
+      } catch (error) {
+        console.error('Error parsing blockedUserList:', error);
+        blockedUserArray = [];
+      }
+    } else {
+      blockedUserArray = blockedUserList;
+    }
+    const blockedUsers = blockedUserArray.map((user) => user.user_id_blocked);
     const reaction = await reactionList(params.id, query.kind, query.limit);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The error handling for the JSON parsing operation could be improved. If an error occurs during parsing, it's currently just logged to the console and the `blockedUserArray` is set to an empty array. This might not be the best approach because if there's a problem with the data format, it would be better to stop the execution and send an appropriate response back to the client. Here's how you can improve this:

```diff
    if (typeof blockedUserList === 'string') {
      try {
        blockedUserArray = JSON.parse(blockedUserList);
      } catch (error) {
-        console.error('Error parsing blockedUserList:', error);
-        blockedUserArray = [];
+        return res.status(400).json({ error: 'Invalid data format' });
      }
    } else {
      blockedUserArray = blockedUserList;
    }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->